cdk/api.py

```diff
@@ -8,15 +8,16 @@ class ZipApi(core.Construct):
 
     def __init__(self, scope: core.Construct, id: str,
             webhook_function,
-            on_demand_function):
+            on_demand_function,
+            lambda_release_alias):
         super().__init__(scope, id)
 
         stack_name = core.Stack.of(self).stack_name
 
         self.rest_api_name = f"{stack_name}-api"
         self.api = apigw.LambdaRestApi(
             self, "api",
-            handler=webhook_function,  # default handler
+            handler=webhook_function.function,  # default handler
             rest_api_name=self.rest_api_name,
```

Comment:
here you're mixing your classes, for example `<class 'cdk.function.ZipWebhookFunction'>` with the cdk classes such as `<class 'aws_cdk.aws_lambda.Function'>`

Reply:
yeah i was doing that in a few other places too. Pycharm helps flag some of these kinds of things but misses a lot too. It kindof makes me miss the stronger typing of typescript. I know you can bring some of that typing into python with the jsii stuff, but not sure how far to take that. Seems un-pythonic. It's also possible there's a way to make pycharm smarter about it's analysis?